lib/execution_engine2/utils/Condor.py

```diff
@@ -137,8 +137,8 @@ def _create_requirements_statement(self, job_reqs: JobRequirements) -> str:
             reqs = [f'regexp("{job_reqs.client_group}",CLIENTGROUP)']
         else:
             reqs = [f'(CLIENTGROUP == "{job_reqs.client_group}")']
-        for key, value in job_reqs.scheduler_requirements.items():
-            reqs.append(f'({key} == "{value}")')
+        for key in sorted(job_reqs.scheduler_requirements):
```

Comment:
Tests passed previously since python dicts maintain their ordering. Since we're now doing full integration tests that go through a JSON-RPC layer we can't rely on that property so we sort the requirements, which we expect to be small. In the extremely unlikely event that this causes a performance problem we can refactor the tests to not rely on sorting.